src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/Server/SqlDataRecord.cs

```diff
@@ -300,162 +89,11 @@ public virtual void SetValue(int ordinal, object value)
-        public virtual void SetSqlGuid(int ordinal, SqlGuid value) => ValueUtilsSmi.SetSqlGuid(_eventSink, _recordBuffer, ordinal, GetSmiMetaData(ordinal), value);
-
-        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient.Server/SqlDataRecord.xml' path='docs/members[@name="SqlDataRecord"]/SetSqlChars/*' />
-        public virtual void SetSqlChars(int ordinal, SqlChars value) => ValueUtilsSmi.SetSqlChars(_eventSink, _recordBuffer, ordinal, GetSmiMetaData(ordinal), value);
-
-        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient.Server/SqlDataRecord.xml' path='docs/members[@name="SqlDataRecord"]/SetSqlBytes/*' />
-        public virtual void SetSqlBytes(int ordinal, SqlBytes value) => ValueUtilsSmi.SetSqlBytes(_eventSink, _recordBuffer, ordinal, GetSmiMetaData(ordinal), value);
-
-        //  SqlDataRecord public API
-        /// <include file='../../../../../../../../doc/snippets/Microsoft.Data.SqlClient.Server/SqlDataRecord.xml' path='docs/members[@name="SqlDataRecord"]/ctor/*' />
-        public SqlDataRecord(params SqlMetaData[] metaData)
-        {
-            // Initial consistency check
-            if (metaData == null)
-            {
-                throw ADP.ArgumentNull(nameof(metaData));
-            }
-
-            _columnMetaData = new SqlMetaData[metaData.Length];
-            _columnSmiMetaData = new SmiExtendedMetaData[metaData.Length];
-            for (int i = 0; i < _columnSmiMetaData.Length; i++)
-            {
-                if (metaData[i] == null)
-                {
-                    throw ADP.ArgumentNull($"{nameof(metaData)}[{i}]");
-                }
-                _columnMetaData[i] = metaData[i];
-                _columnSmiMetaData[i] = MetaDataUtilsSmi.SqlMetaDataToSmiExtendedMetaData(_columnMetaData[i]);
-            }
-
-            _eventSink = new SmiEventSink_Default();
-            _recordBuffer = new MemoryRecordBuffer(_columnSmiMetaData);
-            _usesStringStorageForXml = true;
-            _eventSink.ProcessMessagesAndThrow();
-        }
-
-        //
-        //  SqlDataRecord private members
-        //
-        internal SmiRecordBuffer RecordBuffer => _recordBuffer;
-
-        internal SqlMetaData[] InternalGetMetaData() => _columnMetaData;
-
-        internal SmiExtendedMetaData[] InternalGetSmiMetaData() => _columnSmiMetaData;
-
-        internal SmiExtendedMetaData GetSmiMetaData(int ordinal)
-        {
-            ThrowIfInvalidOrdinal(ordinal);
-            return _columnSmiMetaData[ordinal];
-        }
-
-        internal void ThrowIfInvalidOrdinal(int ordinal)
-        {
-            if (0 > ordinal || FieldCount <= ordinal)
-            {
-                throw ADP.IndexOutOfRange(ordinal);
-            }
-        }
+         private void SetDateTimeOffsetFrameworkSpecific(int ordinal, DateTimeOffset value) => ValueUtilsSmi.SetDateTimeOffset(_eventSink, _recordBuffer, ordinal, GetSmiMetaData(ordinal), value);
```

Comment:
Consider indent and break down the code when it needs horizontal scrolling.

```suggestion
        private void SetDateTimeOffsetFrameworkSpecific(int ordinal, DateTimeOffset value)
            => ValueUtilsSmi.SetDateTimeOffset(_eventSink, _recordBuffer, ordinal, GetSmiMetaData(ordinal), value);
```